diff --git a/backend/docker/docker-entrypoint.sh b/backend/docker/docker-entrypoint.sh
index 71879282..3599dc17 100755
--- a/backend/docker/docker-entrypoint.sh
+++ b/backend/docker/docker-entrypoint.sh
@@ -315,6 +315,10 @@ case "$1" in
       echo "Running celery export worker healthcheck..."
       exec celery -A baserow inspect ping -d "export-worker@$HOSTNAME" -t 10 "${@:2}"
     ;;
+    celery-cloudlanguagetoolsworker)
+      echo "Running celery cloudlanguagetools worker..."
+      start_celery_worker -Q cloudlanguagetools -n cloudlanguagetools-worker@%h "${@:2}"
+    ;;    
     celery-beat)
       # Delay the beat startup as there seems to be bug where the other celery workers
       # starting up interfere with or break the lock obtained by it. Without this the
diff --git a/backend/src/baserow/config/settings/base.py b/backend/src/baserow/config/settings/base.py
index ddaf9f88..4e43ef74 100644
--- a/backend/src/baserow/config/settings/base.py
+++ b/backend/src/baserow/config/settings/base.py
@@ -11,6 +11,23 @@ from corsheaders.defaults import default_headers
 
 from baserow.version import VERSION
 
+# sentry setup
+# ============
+
+import sentry_sdk
+from sentry_sdk.integrations.django import DjangoIntegration
+from sentry_sdk.integrations.celery import CeleryIntegration
+
+sentry_sdk.init(
+    dsn="https://f7a7fa7dfe5f412f852c3bfe2defa091@o968582.ingest.sentry.io/6742581",
+    integrations=[DjangoIntegration(), CeleryIntegration()],
+    traces_sample_rate=1.0,
+    send_default_pii=True,
+)
+
+# sentry setup end
+# ================
+
 BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
 
 BASEROW_PLUGIN_DIR_PATH = Path(os.environ.get("BASEROW_PLUGIN_DIR", "/baserow/plugins"))
@@ -136,6 +153,8 @@ CELERY_TASK_ROUTES = {
     "baserow.core.usage.tasks": {"queue": BASEROW_GROUP_STORAGE_USAGE_QUEUE},
     "baserow.contrib.database.table.tasks.run_row_count_job": {"queue": "export"},
     "baserow.core.jobs.tasks.clean_up_jobs": {"queue": "export"},
+    # cloudlanguagetools jobs go on a separate queue
+    "*cloudlanguagetools*": {"queue": "cloudlanguagetools"},
 }
 CELERY_SOFT_TIME_LIMIT = 60 * 5  # 5 minutes
 CELERY_TIME_LIMIT = CELERY_SOFT_TIME_LIMIT + 60  # 60 seconds
diff --git a/supervisor/default_baserow_env.sh b/supervisor/default_baserow_env.sh
index 41f8dc1b..79644711 100755
--- a/supervisor/default_baserow_env.sh
+++ b/supervisor/default_baserow_env.sh
@@ -74,6 +74,7 @@ export DJANGO_SETTINGS_MODULE=${DJANGO_SETTINGS_MODULE:-$DEFAULT_DJANGO_SETTINGS
 export BASEROW_WEB_FRONTEND_STARTUP_COMMAND="${BASEROW_WEB_FRONTEND_STARTUP_COMMAND:-$DEFAULT_WEB_FRONTEND_STARTUP_COMMAND}"
 export BASEROW_BACKEND_STARTUP_COMMAND="${BASEROW_BACKEND_STARTUP_COMMAND:-$DEFAULT_BACKEND_STARTUP_COMMAND}"
 export BASEROW_CELERY_WORKER_STARTUP_COMMAND="${BASEROW_CELERY_WORKER_STARTUP_COMMAND:-$DEFAULT_CELERY_WORKER_STARTUP_COMMAND}"
+export BASEROW_CELERY_CLT_WORKER_STARTUP_COMMAND="celery-cloudlanguagetoolsworker"
 export BASEROW_CELERY_EXPORT_WORKER_STARTUP_COMMAND="${BASEROW_CELERY_EXPORT_WORKER_STARTUP_COMMAND:-$DEFAULT_CELERY_EXPORT_WORKER_STARTUP_COMMAND}"
 export BASEROW_CELERY_BEAT_STARTUP_COMMAND="${BASEROW_CELERY_BEAT_STARTUP_COMMAND:-$DEFAULT_CELERY_BEAT_STARTUP_COMMAND}"
 export XDG_CONFIG_HOME=/home/$DOCKER_USER/
diff --git a/supervisor/supervisor.conf b/supervisor/supervisor.conf
index 0856cd2f..a2d9f791 100644
--- a/supervisor/supervisor.conf
+++ b/supervisor/supervisor.conf
@@ -60,6 +60,19 @@ priority=2
 startretries=3
 startsecs=30
 
+[program:celerycltworker]
+user=%(ENV_DOCKER_USER)s
+directory=/baserow/backend
+command=/baserow/supervisor/wrapper.sh CYAN CLT_WORKER ./docker/docker-entrypoint.sh %(ENV_BASEROW_CELERY_CLT_WORKER_STARTUP_COMMAND)s
+stdout_logfile=/dev/stdout
+stdout_logfile_maxbytes=0
+stderr_logfile=/dev/stdout
+stderr_logfile_maxbytes=0
+autostart=true
+priority=2
+startretries=3
+startsecs=30
+
 [program:exportworker]
 user=%(ENV_DOCKER_USER)s
 directory=/baserow/backend
diff --git a/web-frontend/config/nuxt.config.base.js b/web-frontend/config/nuxt.config.base.js
index 8a5a4a22..9b12d3c3 100644
--- a/web-frontend/config/nuxt.config.base.js
+++ b/web-frontend/config/nuxt.config.base.js
@@ -19,11 +19,22 @@ export default function (
     base + '/modules/core/module.js',
     base + '/modules/database/module.js',
     premiumBase + '/modules/baserow_premium/module.js',
+    '@nuxtjs/sentry'
   ]
 
   const modules = baseModules.concat(additionalModules)
   return {
     modules,
+    sentry: {
+      dsn: 'https://33f709910b214ed282315bd91344bae0@o968582.ingest.sentry.io/6742673',
+      config: {
+      },
+      publishRelease: {
+        authToken: '9b89e612331511edbbd996b6a33f5072',
+        org: 'language-tools',
+        project: 'baserow-vocabai-frontend',
+      }      
+    },    
     build: {
       extend(config, ctx) {
         config.node = { fs: 'empty' }
diff --git a/web-frontend/package.json b/web-frontend/package.json
index 91dceb77..1ed9a6df 100644
--- a/web-frontend/package.json
+++ b/web-frontend/package.json
@@ -45,7 +45,8 @@
     "uuid": "^8.3.2",
     "vue-chartjs": "^3.5.1",
     "vuejs-datepicker": "^1.6.2",
-    "vuelidate": "^0.7.5"
+    "vuelidate": "^0.7.5",
+    "@nuxtjs/sentry": "^6.0.0"
   },
   "devDependencies": {
     "@nuxtjs/eslint-config": "^6.0.1",
@@ -78,6 +79,7 @@
     "stylelint": "^13.13.1",
     "stylelint-config-standard": "^22.0.0",
     "stylelint-webpack-plugin": "^3.0.1",
-    "vue-jest": "^3.0.3"
+    "vue-jest": "^3.0.3",
+    "@sentry/webpack-plugin": "^1.19.0"
   }
 }
