diff -Naur baserow/web-frontend/config/nuxt.config.base.js baserow-vocabai-patched/web-frontend/config/nuxt.config.base.js
--- baserow/web-frontend/config/nuxt.config.base.js	2022-09-13 11:12:18.240000000 +0800
+++ baserow-vocabai-patched/web-frontend/config/nuxt.config.base.js	2022-09-13 11:39:17.920000000 +0800
@@ -19,11 +19,22 @@
     base + '/modules/core/module.js',
     base + '/modules/database/module.js',
     premiumBase + '/modules/baserow_premium/module.js',
+    '@nuxtjs/sentry'
   ]
 
   const modules = baseModules.concat(additionalModules)
   return {
     modules,
+    sentry: {
+      dsn: 'https://33f709910b214ed282315bd91344bae0@o968582.ingest.sentry.io/6742673',
+      config: {
+      },
+      publishRelease: {
+        authToken: '9b89e612331511edbbd996b6a33f5072',
+        org: 'language-tools',
+        project: 'baserow-vocabai-frontend',
+      }      
+    },    
     build: {
       extend(config, ctx) {
         config.node = { fs: 'empty' }
diff -Naur baserow/web-frontend/package.json baserow-vocabai-patched/web-frontend/package.json
--- baserow/web-frontend/package.json	2022-09-13 11:12:22.940000000 +0800
+++ baserow-vocabai-patched/web-frontend/package.json	2022-09-13 11:35:43.260000000 +0800
@@ -45,7 +45,8 @@
     "uuid": "^8.3.2",
     "vue-chartjs": "^3.5.1",
     "vuejs-datepicker": "^1.6.2",
-    "vuelidate": "^0.7.5"
+    "vuelidate": "^0.7.5",
+    "@nuxtjs/sentry": "^6.0.0"
   },
   "devDependencies": {
     "@nuxtjs/eslint-config": "^6.0.1",
@@ -78,6 +79,7 @@
     "stylelint": "^13.13.1",
     "stylelint-config-standard": "^22.0.0",
     "stylelint-webpack-plugin": "^3.0.1",
-    "vue-jest": "^3.0.3"
+    "vue-jest": "^3.0.3",
+    "@sentry/webpack-plugin": "^1.19.0"
   }
 }
